didx-xyz/yoma-mobile
# YOMA-485 - feat(User): fetch credential data

# Basically done, waiting on api update from G.
## Scope [YOMA-485](https://yomaworld.atlassian.net/browse/YOMA=485)
- fetch user credentials on auth success

![](https://media.giphy.com/media/5G98t8QjqBLK8/giphy.gif)
## Work Done
- adding actions and middleware to fetch the user credentials in User module
- adding action and middleware to hydrate the app in App module
- dispatched app hydrate action on Auth success.
- added `UserCredentialTypes` enum to handle different credential types
- added hydrateApp action that runs on authorization of the user.
- moved the `UserResponse` type from `Auth` to `User` module

### Additional Work: 
- ran into some testing issues, so attempted to fix those by:
  - removing SSO functions (as that's being refactored anyway)
  - adding jest.setup file and adjusted some of the other test mocks
- renamed `createMiddlewareMock` to `createMiddlewareStub` (this affected a few other test files as well)
- refactored and renamed `api util` `addIdBeforeEndpointInConfig` to `prependIdToEndpointInConfig`
- fixed `rootStateFixture` to properly accept an empty override argument. 
- adding `selectUserId` selector (might end up merging as it's done in other PRs already)
- adding user test fixture

## Steps to Test
- run the app and sign in - and see that the credentials are successfully returned in the payload for `fetchUserCredentialsSuccess`
